### PR TITLE
[live555] Fix install head failed

### LINKED ISSUE #17165: [live555] build failure
**Host Environment**
- OS: [Windows]
- Compiler: MSVC 19.16.27045

**To Reproduce**
`./vcpkg install live555 --head`

**Failure logs** 
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6282509/install-x64-windows-dbg-out.log)

**Additional context**
none

### PR DESCRIPTION
**Pull request by JonLiu1993**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #17165

live555 install failed whit --head, Add this condition
````
if (WIN32)
    add_definitions(-DNO_GETIFADDRS)
endif(WIN32)
````
Fix the issue

### CODE REVIEW

**Review by JackBoosY — CHANGES_REQUESTED** [review]
I think we should also update this port, can you?

File: ports/live555/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,10 @@ include_directories(
     UsageEnvironment/include
 )
 
+if (WIN32)
+    add_definitions(-DNO_GETIFADDRS)
+endif(WIN32)
```
Comment (JackBoosY) [inline ports/live555/CMakeLists.txt:13]: In _groupsock/GroupsockHelper.cpp_ line 45:
```cpp
#ifndef NO_GETIFADDRS
#include <ifaddrs.h>
#include <net/if.h>
#endif
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]
```cmake
vcpkg_download_distfile(ARCHIVE
    URLS "http://www.live555.com/liveMedia/public/live.2021.12.18.tar.gz"
    FILENAME "live.2021.12.18.tar.gz"
    SHA512 072ae92a1a63687e33c659627fbf562e6f2cdf73cf1c3199959aa3a713636ccd4113d56270c4793c86b42338000f180e16a7b2cf3d893402226a04360f8b2e97
)
```

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update, and investigation of the build failure!

**Pull request merged**